src/main/scala/eventstore/StreamSubscriptionActor.scala

```diff
@@ -163,7 +163,7 @@ class StreamSubscriptionActor private (
       }
 
       def subscribed(number: Last) = {
-        catchUp(subscriptionNumber, Queue())
+        catchUp(number.getOrElse(subscriptionNumber), Queue())
```

Comment:
I guess that we should pass the current stash instead of a blank queue here.

Reply:
No, this is ok, because we don't need events in the stash because we will catch up to the new subscription number.